mmengine/hooks/freeze_hook.py

```diff
@@ -0,0 +1,121 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from typing import Optional, Sequence, Union
+
+from mmengine.logging import print_log
+from mmengine.model import is_model_wrapper
+from mmengine.registry import HOOKS
+from .hook import Hook
+
+
+@HOOKS.register_module()
+class FreezeHook(Hook):
+    """FreezeHook is used to freeze or unfreeze network layers when training to
+    a specified epoch.
+
+    Args:
+        freeze_epoch (int): The epoch number to start freezing layers.
+        freeze_layers (tuple[str]): Model layers containing the keyword in
```

Comment:
Suggest making `freeze_layers` the first argument, and it should be a regex expression